packages/venia-concept/src/components/Checkout/paymentsForm.js

```diff
@@ -22,228 +21,229 @@ const DEFAULT_FORM_VALUES = {
+    // then modified when using initialValues any component who uses this
+    // callback will be rendered multiple times on first render. See above
+    // comments for more info.
+    const handleSuccess = useCallback(
+        value => {
+            setIsSubmitting(false);
+            const sameAsShippingAddress = formState.values['addresses_same'];
+            let billingAddress;
+            if (!sameAsShippingAddress) {
+                billingAddress = {
+                    city: formState.values['city'],
+                    postcode: formState.values['postcode'],
+                    region_code: formState.values['region_code'],
+                    street: formState.values['street']
                 };
             } else {
-                // The addresses are not the same, populate the other fields.
-                initialFormValues = {
-                    addresses_same: false,
-                    ...initialValues
+                billingAddress = {
+                    sameAsShippingAddress
                 };
-                delete initialFormValues.sameAsShippingAddress;
             }
-        }
-
-        return (
-            <Form
-                className={classes.root}
-                getApi={this.setFormApi}
-                initialValues={initialFormValues}
-                onSubmit={this.submit}
-            >
-                {formChildren}
-            </Form>
-        );
-    }
-
-    /*
-     *  Class Properties.
-     */
-    billingAddressFields = () => {
-        const { classes, countries } = this.props;
-
-        return (
-            <Fragment>
-                <div className={classes.street0}>
-                    <Field label="Street">
-                        <TextInput
-                            id={classes.street0}
-                            field="street[0]"
-                            validate={isRequired}
-                        />
-                    </Field>
-                </div>
-                <div className={classes.city}>
-                    <Field label="City">
-                        <TextInput
-                            id={classes.city}
-                            field="city"
-                            validate={isRequired}
-                        />
-                    </Field>
+            submit({
+                billingAddress,
+                paymentMethod: {
+                    code: 'braintree',
+                    data: value
+                }
+            });
+        },
+        [formState.values]
+    );
+
+    return (
+        <Fragment>
+            <div className={classes.body}>
+                <h2 className={classes.heading}>Billing Information</h2>
+                <div className={classes.braintree}>
+                    <BraintreeDropin
+                        shouldRequestPaymentNonce={isSubmitting}
+                        onError={handleError}
+                        onSuccess={handleSuccess}
+                    />
```

Comment:
Is there any way for `BraintreeDropin` to be loaded with `React.lazy()` and `Suspense`?

Reply:
I can investigate. TBH I haven't used either yet, so I'm eager to try.

Reply:
Do you still want me to look into this? Not sure how much work it'll be do we want to add to this PR?